File: leetsync/2260-divide-a-string-into-groups-of-size-k/divide-a-string-into-groups-of-size-k.go
func divideString(s string, k int, fill byte) []string {
	for len(s)%k != 0 {
		s += string(fill)
	}
	n := len(s)
	var result []string
	for i := 0; i < n; i += k {
		result = append(result, s[i:i+k])
	}
	return result
}

File: leetsync/2260-divide-a-string-into-groups-of-size-k/divide-a-string-into-groups-of-size-k.java
class Solution {
    public String[] divideString(String s, int k, char fill) {
        while (s.length() % k != 0) {
            s += fill;
        }
        int n = s.length();
        String[] result = new String[n / k];
        int i = 0, index = 0;
        while (i < n) {
            result[index++] = s.substring(i, i + k);
            i += k;
        }
        return result;
    }
}